### PR TITLE
Two tests didn't properly assert an exception was raised. Fixed.

### PR DESCRIPTION
**Pull request by akaihola**
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.24.0.txt [hunk 1]
```diff
@@ -191,6 +191,5 @@ Other
 ^^^^^
 
 - :meth: `~pandas.io.formats.style.Styler.background_gradient` now takes a ``text_color_threshold`` parameter to automatically lighten the text color based on the luminance of the background color. This improves readability with dark background colors without the need to limit the background colormap range. (:issue:`21258`)
--
--
+- fixed two test cases which asserted an exception message but failed to make sure that an exception was actually raised
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: not necessary for test changes

File: pandas/tests/io/parser/c_parser_only.py [hunk 2]
```diff
@@ -34,10 +34,9 @@ def test_buffer_overflow(self):
         cperr = 'Buffer overflow caught - possible malformed input file.'
 
         for malf in (malfw, malfs, malfl):
-            try:
+            with pytest.raises(pd.errors.ParserError) as excinfo:
```
Comment (jreback) [inline pandas/tests/io/parser/c_parser_only.py]: can u parametrize this test

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.24.0.txt (hunk 1 above)
Comment (akaihola) [reply doc/source/whatsnew/v0.24.0.txt]: Ok, reverted.

File: pandas/tests/io/parser/c_parser_only.py (hunk 2 above)
Comment (akaihola) [reply pandas/tests/io/parser/c_parser_only.py]: Parametrized in f47437e.

**Review by gfyoung — APPROVED** [review]
Nice catch!

cc @jreback

**Comment by jreback** [comment]
thanks @akaihola; slight change to use pytest ids, otherwise nice catch! keep em coming!

**Pull request merged**